phantom-sbt/src/main/scala/com/websudos/phantom/sbt/SbtPlugin.scala

```diff
@@ -0,0 +1,149 @@
+import sbt.plugins.JvmPlugin
+
+import scala.concurrent.blocking
+import scala.util.control.NonFatal
+
+/**
+  * sbt plugin for starting Cassandra in embedded mode before running
+  * the tests.
+  *
+  * First the plugin must be included in your `plugins.sbt`:
+  *
+  * {{{
+  *   addSbtPlugin("com.websudos" %% "phantom-sbt" % phantomVersion)
+  * }}}
+  * The plugin does the following
+  * things:
+  *
+  * - Automatically starts Cassandra in embedded mode whenever the test task is run
+  * - Forces the tests for the projects that include the settings to always run in a
+  *   forked JVM as this is the only way to make parallel tests using phantom work.
+  *   (This is not caused by the implementation of this plugin or the new connector
+  *   or zookeeper artifacts, this is caused by implementation details in the official
+  *   `phantom-dsl` artifact, mainly the use of reflection which is not thread-safe
+  *   in Scala 2.10)
+  *
+  * If you want to specify a custom Cassandra configuration,
+  * you can do that with a setting:
+  *
+  * {{{
+  * phantomCassandraConfig := baseDirectory.value / "config" / "cassandra.yaml"
+  * }}}
+  */
+object PhantomSbtPlugin extends AutoPlugin {
+
+  override def requires: Plugins = sbt.plugins.JvmPlugin
+
+  /**
+    * Keys for all settings of this plugin.
+    */
+  object autoImport {
+
+    val phantomStartEmbeddedCassandra = taskKey[Unit]("Starts embedded Cassandra")
+    val phantomCleanupEmbeddedCassandra = taskKey[Unit]("Clean up embedded Cassandra by dropping all of its keyspaces")
+
+    val phantomCassandraConfig = settingKey[Option[File]]("YAML file for Cassandra configuration")
+  }
+
+  import autoImport._
+
+  override def projectSettings: Seq[Def.Setting[_]] = Seq(
+    phantomCassandraConfig := None,
+    phantomStartEmbeddedCassandra := EmbeddedCassandra.start(phantomCassandraConfig.value, streams.value.log),
+    phantomCleanupEmbeddedCassandra := EmbeddedCassandra.cleanup(streams.value.log),
+    test in Test <<= (test in Test).dependsOn(phantomStartEmbeddedCassandra),
+    testQuick in Test <<= (testQuick in Test).dependsOn(phantomStartEmbeddedCassandra),
+    testOnly in Test <<= (testOnly in Test).dependsOn(phantomStartEmbeddedCassandra),
+    fork := true
+  )
+}
+
+/**
+  * Singleton object that is responsible for starting
+  * Cassandra in embedded mode, but only once.
+  * Subsequent calls to `start` will be ignored.
+  */
+object EmbeddedCassandra {
+
+  private[this] var started: Boolean = false
+
+  /**
+    * Starts Cassandra in embedded mode if it has not been
+    * started yet.
+    */
+  def start (config: Option[File], logger: Logger): Unit = {
+    logger.info("Initialising EmbeddedCassandra singleton.")
+    this.synchronized {
+      if (!started) {
+        blocking {
+          val configFile = config.map(_.toURI.toString) getOrElse EmbeddedCassandraServerHelper.DEFAULT_CASSANDRA_YML_FILE
+          System.setProperty("cassandra.config", configFile)
+          try {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Consider using util.Try instead of a try-catch block](https://www.codacy.com/app/flavian/phantom/file/2520108775/issues/source?bid=2847375&fileBranchId=3265057#l115)